models/Category.php

```diff
@@ -64,14 +66,18 @@ public function getPostCountAttribute()
 
     /**
      * Sets the "url" attribute with a URL to this object
+     *
      * @param string $pageName
-     * @param Cms\Classes\Controller $controller
+     * @param Controller $controller
+     * @param array $params
+     *
+     * @return string
      */
-    public function setUrl($pageName, $controller)
+    public function setUrl($pageName, $controller, array $params = array())
     {
         $params = [
-            'id'   => $this->id,
-            'slug' => $this->slug,
+            $this->getModelUrlParam('id', $params)   => $this->id,
```

Comment:
Can't we avoid the extra trait entirely just by using `array_get()` instead? https://laravel.com/docs/5.6/helpers#method-array-get

```php
array_get($params, 'id', 'id');
```

Reply:
Indeed, nice one